splash/resources.py

```diff
@@ -102,6 +105,7 @@ def _writeOutput(self, data, request, content_type=None):
             content_type = self.content_type
 
         if isinstance(data, (dict, list)):
+            data = bytes_to_unicode(data)
```

Comment:
data almost always contains bytes because lupa always returns bytes, not unicode.